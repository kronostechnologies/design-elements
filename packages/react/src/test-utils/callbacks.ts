export function doNothing(): void {
    // do nothing
}
