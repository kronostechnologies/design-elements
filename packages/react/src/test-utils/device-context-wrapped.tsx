import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
