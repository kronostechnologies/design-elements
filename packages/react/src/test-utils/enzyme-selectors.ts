import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
