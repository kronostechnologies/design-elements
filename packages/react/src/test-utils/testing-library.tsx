import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { DesignSystem, DesignSystemProps } from '../components/design-system';
import { equisoftTheme } from '../theme';

export function renderWithProviders(
    ui: Parameters<typeof render>[0],
    renderOptions: Omit<Parameters<typeof render>[1], 'wrapper'> | undefined = undefined,
    wrapperProps: DesignSystemProps = {},
): ReturnType<typeof render> {
    const {
        isolateStyles,
        language,
        staticDevice,
        theme,
    } = {
        isolateStyles: false,
        language: 'fr',
        theme: equisoftTheme,
        ...wrapperProps,
    };
    return render(
        ui,
        {
            ...renderOptions,
            wrapper: ({ children }) => (
                <MemoryRouter future={{ v7_startTransition: false, v7_relativeSplatPath: false }}>
                    <DesignSystem
                        isolateStyles={isolateStyles}
                        language={language}
                        staticDevice={staticDevice}
                        theme={theme}
                    >
                        {children}
                    </DesignSystem>
                </MemoryRouter>
            ),
        },
    );
}
