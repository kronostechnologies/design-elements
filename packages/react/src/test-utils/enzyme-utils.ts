import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

export async function waitForComponentToPaint<P = {}>(
    wrapper: ReactWrapper<P>,
    amount = 0,
): Promise<void> {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, amount));
        wrapper.update();
    });
}

export function expectFocusToBeOn(element: ReactWrapper): void {
    expect(document.activeElement).toBe(element.getDOMNode());
}
