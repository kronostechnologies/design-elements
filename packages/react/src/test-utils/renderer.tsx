import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
