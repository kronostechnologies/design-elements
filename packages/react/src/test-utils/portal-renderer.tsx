import { render, RenderResult } from '@testing-library/react';
import React, { ReactElement } from 'react';
import { DeviceType } from '../components/device-context-provider/device-context-provider';
import { AllProviders } from './renderer';

export function renderPortalWithProviders(
    component: ReactElement,
    device?: DeviceType,
): RenderResult {
    return render(<AllProviders device={device}>{component}</AllProviders>);
}
