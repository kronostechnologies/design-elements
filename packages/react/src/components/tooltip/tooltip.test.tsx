import { renderWithProviders } from '@design-elements/test-utils/renderer';
import React from 'react';
import { Tooltip, TooltipArrow, TooltipContainer } from './tooltip';

jest.mock('uuid/v4');

describe('Tooltip', () => {
    test('Has default desktop styles', () => {
        const tree = renderWithProviders(
            <Tooltip>
                Test Content
            </Tooltip>,
        );

        expect(tree).toMatchSnapshot();
    });

    test('Has mobile styles', () => {
        const tree = renderWithProviders(
            <Tooltip>
                Test Content
            </Tooltip>,
            'mobile',
        );

        expect(tree).toMatchSnapshot();
    });

    test('TooltipContainer has desktop styles', () => {
        const tree = renderWithProviders(
            <TooltipContainer>
                <TooltipArrow />
                Test Content
            </TooltipContainer>,
        );

        expect(tree).toMatchSnapshot();
    });

    test('TooltipContainer has mobile styles', () => {
        const tree = renderWithProviders(
            <TooltipContainer isMobile>
                <TooltipArrow />
                Test Content
            </TooltipContainer>,
        );

        expect(tree).toMatchSnapshot();
    });
});
