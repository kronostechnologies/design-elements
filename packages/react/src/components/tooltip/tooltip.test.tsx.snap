// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Has default desktop styles (defaultOpen) 1`] = `
.c3 {
  height: 1.25rem;
  position: absolute;
  width: 1rem;
}

.c3::before {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  width: 0;
}

.c3::after {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  position: absolute;
  width: 0;
}

.c1 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius-half);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  margin: 0;
  max-width: 327px;
  min-height: 24px;
  padding: 1px var(--spacing-1x);
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  z-index: 1000;
}

.c1[data-popper-placement*="bottom"] > .c2 {
  height: 1rem;
  left: 0;
  margin-top: -0.375rem;
  top: 0;
  width: 1rem;
}

.c1[data-popper-placement*="bottom"] > .c2::before {
  border-color: transparent transparent #FFFFFF transparent;
  border-width: 0 0.5rem 0.5rem;
  position: absolute;
  top: -2px;
}

.c1[data-popper-placement*="bottom"] > .c2::after {
  border-color: transparent transparent #60666E transparent;
  border-width: 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="top"] > .c2 {
  bottom: 0;
  height: 0;
  left: 0;
  margin-bottom: 0;
  width: 1rem;
}

.c1[data-popper-placement*="top"] > .c2::before {
  border-color: #FFFFFF transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  position: absolute;
  top: 0;
}

.c1[data-popper-placement*="top"] > .c2::after {
  border-color: #60666E transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  top: -0.1rem;
}

.c1[data-popper-placement*="right"] > .c2 {
  height: 1rem;
  left: 0;
  margin-left: -0.8rem;
  width: 1rem;
}

.c1[data-popper-placement*="right"] > .c2::before {
  border-color: transparent #FFFFFF transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
}

.c1[data-popper-placement*="right"] > .c2::after {
  border-color: transparent #60666E transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
  left: 0.375rem;
  top: 0;
}

.c1[data-popper-placement*="left"] > .c2 {
  height: 1rem;
  margin-right: -0.7rem;
  right: -1px;
  width: 1rem;
}

.c1[data-popper-placement*="left"] > .c2::before {
  border-color: transparent transparent transparent #FFFFFF;
  border-width: 0.5rem 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="left"] > .c2::after {
  border-color: transparent transparent transparent #60666E;
  border-width: 0.5rem 0 0.5rem 0.5rem;
  left: 0.17rem;
  top: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

.c0:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

<Tooltip
  defaultOpen={true}
  label="Test Content"
>
  <styled.span
    aria-describedby="uuid1"
    data-testid="tooltip"
    id="tooltip-trigger-uuid1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-describedby="uuid1"
      className="c0"
      data-testid="tooltip"
      id="tooltip-trigger-uuid1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      tabIndex={0}
    >
      <Icon
        color="#60666E"
        name="info"
        size="16"
      >
        <svg
          color="#60666E"
          focusable={false}
          height="16"
          width="16"
        />
      </Icon>
    </span>
  </styled.span>
  <styled.div
    aria-hidden={false}
    data-popper-interactive={false}
    data-testid="tooltip-content-container"
    id="uuid1"
    isMobile={false}
    role="tooltip"
    style={
      {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
    variant="normal"
    visible={true}
  >
    <div
      aria-hidden={false}
      className="c1"
      data-popper-interactive={false}
      data-testid="tooltip-content-container"
      id="uuid1"
      role="tooltip"
      style={
        {
          "left": "0",
          "position": "absolute",
          "top": "0",
        }
      }
    >
      <styled.div
        data-popper-arrow={true}
        style={
          {
            "position": "absolute",
          }
        }
      >
        <div
          className="c2 c3"
          data-popper-arrow={true}
          style={
            {
              "position": "absolute",
            }
          }
        />
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          Test Content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Tooltip>
`;

exports[`Tooltip Has default desktop styles 1`] = `
.c3 {
  height: 1.25rem;
  position: absolute;
  width: 1rem;
}

.c3::before {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  width: 0;
}

.c3::after {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  position: absolute;
  width: 0;
}

.c1 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius-half);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 0.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  margin: 0;
  max-width: 327px;
  min-height: 24px;
  padding: 1px var(--spacing-1x);
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  z-index: 1000;
}

.c1[data-popper-placement*="bottom"] > .c2 {
  height: 1rem;
  left: 0;
  margin-top: -0.375rem;
  top: 0;
  width: 1rem;
}

.c1[data-popper-placement*="bottom"] > .c2::before {
  border-color: transparent transparent #FFFFFF transparent;
  border-width: 0 0.5rem 0.5rem;
  position: absolute;
  top: -2px;
}

.c1[data-popper-placement*="bottom"] > .c2::after {
  border-color: transparent transparent #60666E transparent;
  border-width: 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="top"] > .c2 {
  bottom: 0;
  height: 0;
  left: 0;
  margin-bottom: 0;
  width: 1rem;
}

.c1[data-popper-placement*="top"] > .c2::before {
  border-color: #FFFFFF transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  position: absolute;
  top: 0;
}

.c1[data-popper-placement*="top"] > .c2::after {
  border-color: #60666E transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  top: -0.1rem;
}

.c1[data-popper-placement*="right"] > .c2 {
  height: 1rem;
  left: 0;
  margin-left: -0.8rem;
  width: 1rem;
}

.c1[data-popper-placement*="right"] > .c2::before {
  border-color: transparent #FFFFFF transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
}

.c1[data-popper-placement*="right"] > .c2::after {
  border-color: transparent #60666E transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
  left: 0.375rem;
  top: 0;
}

.c1[data-popper-placement*="left"] > .c2 {
  height: 1rem;
  margin-right: -0.7rem;
  right: -1px;
  width: 1rem;
}

.c1[data-popper-placement*="left"] > .c2::before {
  border-color: transparent transparent transparent #FFFFFF;
  border-width: 0.5rem 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="left"] > .c2::after {
  border-color: transparent transparent transparent #60666E;
  border-width: 0.5rem 0 0.5rem 0.5rem;
  left: 0.17rem;
  top: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

.c0:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

<Tooltip
  label="Test Content"
>
  <styled.span
    aria-describedby="uuid1"
    data-testid="tooltip"
    id="tooltip-trigger-uuid1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-describedby="uuid1"
      className="c0"
      data-testid="tooltip"
      id="tooltip-trigger-uuid1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      tabIndex={0}
    >
      <Icon
        color="#60666E"
        name="info"
        size="16"
      >
        <svg
          color="#60666E"
          focusable={false}
          height="16"
          width="16"
        />
      </Icon>
    </span>
  </styled.span>
  <styled.div
    aria-hidden={true}
    data-popper-interactive={false}
    data-testid="tooltip-content-container"
    id="uuid1"
    isMobile={false}
    role="tooltip"
    style={
      {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
    variant="normal"
    visible={false}
  >
    <div
      aria-hidden={true}
      className="c1"
      data-popper-interactive={false}
      data-testid="tooltip-content-container"
      id="uuid1"
      role="tooltip"
      style={
        {
          "left": "0",
          "position": "absolute",
          "top": "0",
        }
      }
    >
      <styled.div
        data-popper-arrow={true}
        style={
          {
            "position": "absolute",
          }
        }
      >
        <div
          className="c2 c3"
          data-popper-arrow={true}
          style={
            {
              "position": "absolute",
            }
          }
        />
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          Test Content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Tooltip>
`;

exports[`Tooltip Has mobile styles (defaultOpen) 1`] = `
.c3 {
  height: 1.25rem;
  position: absolute;
  width: 1rem;
}

.c3::before {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  width: 0;
}

.c3::after {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  position: absolute;
  width: 0;
}

.c1 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius-half);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.5rem;
  margin: 0;
  max-width: 327px;
  min-height: 72px;
  padding: 1px var(--spacing-3x);
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  z-index: 1000;
}

.c1[data-popper-placement*="bottom"] > .c2 {
  height: 1rem;
  left: 0;
  margin-top: -0.375rem;
  top: 0;
  width: 1rem;
}

.c1[data-popper-placement*="bottom"] > .c2::before {
  border-color: transparent transparent #FFFFFF transparent;
  border-width: 0 0.5rem 0.5rem;
  position: absolute;
  top: -2px;
}

.c1[data-popper-placement*="bottom"] > .c2::after {
  border-color: transparent transparent #60666E transparent;
  border-width: 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="top"] > .c2 {
  bottom: 0;
  height: 0;
  left: 0;
  margin-bottom: 0;
  width: 1rem;
}

.c1[data-popper-placement*="top"] > .c2::before {
  border-color: #FFFFFF transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  position: absolute;
  top: 0;
}

.c1[data-popper-placement*="top"] > .c2::after {
  border-color: #60666E transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  top: -0.1rem;
}

.c1[data-popper-placement*="right"] > .c2 {
  height: 1rem;
  left: 0;
  margin-left: -0.8rem;
  width: 1rem;
}

.c1[data-popper-placement*="right"] > .c2::before {
  border-color: transparent #FFFFFF transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
}

.c1[data-popper-placement*="right"] > .c2::after {
  border-color: transparent #60666E transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
  left: 0.375rem;
  top: 0;
}

.c1[data-popper-placement*="left"] > .c2 {
  height: 1rem;
  margin-right: -0.7rem;
  right: -1px;
  width: 1rem;
}

.c1[data-popper-placement*="left"] > .c2::before {
  border-color: transparent transparent transparent #FFFFFF;
  border-width: 0.5rem 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="left"] > .c2::after {
  border-color: transparent transparent transparent #60666E;
  border-width: 0.5rem 0 0.5rem 0.5rem;
  left: 0.17rem;
  top: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

.c0:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

<Tooltip
  defaultOpen={true}
  label="Test Content"
>
  <styled.span
    aria-describedby="uuid1"
    data-testid="tooltip"
    id="tooltip-trigger-uuid1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-describedby="uuid1"
      className="c0"
      data-testid="tooltip"
      id="tooltip-trigger-uuid1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      tabIndex={0}
    >
      <Icon
        color="#60666E"
        name="info"
        size="24"
      >
        <svg
          color="#60666E"
          focusable={false}
          height="24"
          width="24"
        />
      </Icon>
    </span>
  </styled.span>
  <styled.div
    aria-hidden={false}
    data-popper-interactive={false}
    data-testid="tooltip-content-container"
    id="uuid1"
    isMobile={true}
    role="tooltip"
    style={
      {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
    variant="normal"
    visible={true}
  >
    <div
      aria-hidden={false}
      className="c1"
      data-popper-interactive={false}
      data-testid="tooltip-content-container"
      id="uuid1"
      role="tooltip"
      style={
        {
          "left": "0",
          "position": "absolute",
          "top": "0",
        }
      }
    >
      <styled.div
        data-popper-arrow={true}
        style={
          {
            "position": "absolute",
          }
        }
      >
        <div
          className="c2 c3"
          data-popper-arrow={true}
          style={
            {
              "position": "absolute",
            }
          }
        />
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          Test Content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Tooltip>
`;

exports[`Tooltip Has mobile styles 1`] = `
.c3 {
  height: 1.25rem;
  position: absolute;
  width: 1rem;
}

.c3::before {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  width: 0;
}

.c3::after {
  border-style: solid;
  content: '';
  display: block;
  height: 0;
  margin: auto;
  position: absolute;
  width: 0;
}

.c1 {
  background-color: #60666E;
  border: 1px solid #FFFFFF;
  border-radius: var(--border-radius-half);
  box-shadow: 0 10px 20px 0 rgb(0 0 0 / 19%);
  box-sizing: border-box;
  color: #FFFFFF;
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.5rem;
  margin: 0;
  max-width: 327px;
  min-height: 72px;
  padding: 1px var(--spacing-3x);
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
  z-index: 1000;
}

.c1[data-popper-placement*="bottom"] > .c2 {
  height: 1rem;
  left: 0;
  margin-top: -0.375rem;
  top: 0;
  width: 1rem;
}

.c1[data-popper-placement*="bottom"] > .c2::before {
  border-color: transparent transparent #FFFFFF transparent;
  border-width: 0 0.5rem 0.5rem;
  position: absolute;
  top: -2px;
}

.c1[data-popper-placement*="bottom"] > .c2::after {
  border-color: transparent transparent #60666E transparent;
  border-width: 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="top"] > .c2 {
  bottom: 0;
  height: 0;
  left: 0;
  margin-bottom: 0;
  width: 1rem;
}

.c1[data-popper-placement*="top"] > .c2::before {
  border-color: #FFFFFF transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  position: absolute;
  top: 0;
}

.c1[data-popper-placement*="top"] > .c2::after {
  border-color: #60666E transparent transparent transparent;
  border-width: 0.5rem 0.5rem 0;
  top: -0.1rem;
}

.c1[data-popper-placement*="right"] > .c2 {
  height: 1rem;
  left: 0;
  margin-left: -0.8rem;
  width: 1rem;
}

.c1[data-popper-placement*="right"] > .c2::before {
  border-color: transparent #FFFFFF transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
}

.c1[data-popper-placement*="right"] > .c2::after {
  border-color: transparent #60666E transparent transparent;
  border-width: 0.5rem 0.5rem 0.5rem 0;
  left: 0.375rem;
  top: 0;
}

.c1[data-popper-placement*="left"] > .c2 {
  height: 1rem;
  margin-right: -0.7rem;
  right: -1px;
  width: 1rem;
}

.c1[data-popper-placement*="left"] > .c2::before {
  border-color: transparent transparent transparent #FFFFFF;
  border-width: 0.5rem 0 0.5rem 0.5rem;
}

.c1[data-popper-placement*="left"] > .c2::after {
  border-color: transparent transparent transparent #60666E;
  border-width: 0.5rem 0 0.5rem 0.5rem;
  left: 0.17rem;
  top: 0;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c0 {
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

.c0:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
  -webkit-transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
  transition: outline .25s ease-in-out,box-shadow .25s ease-in-out;
}

<Tooltip
  label="Test Content"
>
  <styled.span
    aria-describedby="uuid1"
    data-testid="tooltip"
    id="tooltip-trigger-uuid1"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    tabIndex={0}
  >
    <span
      aria-describedby="uuid1"
      className="c0"
      data-testid="tooltip"
      id="tooltip-trigger-uuid1"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      tabIndex={0}
    >
      <Icon
        color="#60666E"
        name="info"
        size="24"
      >
        <svg
          color="#60666E"
          focusable={false}
          height="24"
          width="24"
        />
      </Icon>
    </span>
  </styled.span>
  <styled.div
    aria-hidden={true}
    data-popper-interactive={false}
    data-testid="tooltip-content-container"
    id="uuid1"
    isMobile={true}
    role="tooltip"
    style={
      {
        "left": "0",
        "position": "absolute",
        "top": "0",
      }
    }
    variant="normal"
    visible={false}
  >
    <div
      aria-hidden={true}
      className="c1"
      data-popper-interactive={false}
      data-testid="tooltip-content-container"
      id="uuid1"
      role="tooltip"
      style={
        {
          "left": "0",
          "position": "absolute",
          "top": "0",
        }
      }
    >
      <styled.div
        data-popper-arrow={true}
        style={
          {
            "position": "absolute",
          }
        }
      >
        <div
          className="c2 c3"
          data-popper-arrow={true}
          style={
            {
              "position": "absolute",
            }
          }
        />
      </styled.div>
      <styled.div>
        <div
          className="c4"
        >
          Test Content
        </div>
      </styled.div>
    </div>
  </styled.div>
</Tooltip>
`;
